### ISSUE TITLE
Error when running a KROHF calculation 

### ISSUE THREAD

**Issue by hungpham2017** [pyscf/pyscf#253]
Hello,
I am trying to run the below KROHF calculation using both PySCF 1.5.14 or PySCF from the dev branch and python 3.6.6: 

```
import numpy as np
from pyscf.pbc import gto, scf

cell = gto.Cell()
cell.atom = '''
He 0 0 1
He 1 0 1
'''
cell.basis = '321g'
cell.a = np.eye(3) * 3
cell.mesh = [11] * 3
cell.verbose = 5
cell.spin = 2
cell.build()
nk = [1, 1, 3]
kpts = cell.make_kpts(nk,wrap_around=True)
mf = scf.KROHF(cell, kpts)
mf.kernel()
```

and get this error:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/gagliard/phamx494/pyscf/pyscf/scf/hf.py", line 1476, in kernel
    return self.scf(dm0, **kwargs)
  File "/home/gagliard/phamx494/pyscf/pyscf/scf/hf.py", line 1470, in scf
    conv_check=self.conv_check, **kwargs)
  File "/home/gagliard/phamx494/pyscf/pyscf/scf/hf.py", line 143, in kernel
    vhf = mf.get_veff(mol, dm)
  File "/home/gagliard/phamx494/pyscf/pyscf/pbc/scf/krohf.py", line 335, in get_veff
    vj, vk = self.get_jk(cell, dm_kpts, hermi, kpts, kpts_band)
  File "/home/gagliard/phamx494/pyscf/pyscf/pbc/scf/khf.py", line 493, in get_jk
    exxdiv=self.exxdiv)
  File "/home/gagliard/phamx494/pyscf/pyscf/pbc/df/fft.py", line 288, in get_jk
    vj = fft_jk.get_j_kpts(self, dm, hermi, kpts, kpts_band)
  File "/home/gagliard/phamx494/pyscf/pyscf/pbc/df/fft_jk.py", line 67, in get_j_kpts
    rhoR[i,p0:p1] += make_rho(i, ao_ks, mask, 'LDA')
  File "/home/gagliard/phamx494/pyscf/pyscf/pbc/dft/numint.py", line 1222, in make_rho
    hermi=1)
  File "/home/gagliard/phamx494/pyscf/pyscf/pbc/dft/numint.py", line 1086, in eval_rho
    rhoR += eval_rho(cell, ao_kpts[k], dm_kpts[k], non0tab, xctype,
IndexError: index 2 is out of bounds for axis 0 with size 2
```
Is there anything wrong in the input?
or what could be possible reasons?

Thank you,
Hung

**Comment by hungpham2017** [comment]
You are correct. By inspecting that.
ao_kpts : (3,1331,4) and dm_kpts : (2,4,4)

The **_gen_rho_evaluator** function (**line 1200** of pbc/dft/numint.py) takes a dms (a (2,3,4,4) array) rather than a dm_kpts (a (3,2,4,4) array).

It started at **line 55** of pbc/df/fft_jk.py:
`make_rho, nset, nao = ni._gen_rho_evaluator(cell, dm_kpts, hermi)`

**Comment by gkc1000** [comment]
This requires a bug fix, but a workaround should be the following: this is
using FFTDF, which requires evaluation of rho on a grid etc., which seems
here not to handle the ROHF case properly.
Try using Gaussian DF (should be much faster once the integrals are
evaluated anyways).

Garnet

**Comment by hungpham2017** [comment]
I ran the KROHF calculation using GDF:

```
import numpy as np
from pyscf.pbc import gto, scf, df

cell = gto.Cell()
cell.atom = '''
He 0 0 1
He 1 0 1
'''
cell.basis = '321g'
cell.a = np.eye(3) * 3
cell.mesh = [11] * 3
cell.verbose = 5
cell.build()
nk = [1, 1, 3]
kpts = cell.make_kpts(nk,wrap_around=True)
gdf = df.GDF(cell, kpts)
mf = scf.KROHF(cell, kpts)
mf.with_df = gdf
mf.kernel()
```
and get this:
```
Traceback (most recent call last):
  File "h2.py", line 22, in <module>
    mf.kernel()
  File "/home/gagliard/phamx494/pyscf/pyscf/scf/hf.py", line 1476, in kernel
    return self.scf(dm0, **kwargs)
  File "/home/gagliard/phamx494/pyscf/pyscf/scf/hf.py", line 1470, in scf
    conv_check=self.conv_check, **kwargs)
  File "/home/gagliard/phamx494/pyscf/pyscf/scf/hf.py", line 144, in kernel
    e_tot = mf.energy_tot(dm, h1e, vhf)
  File "/home/gagliard/phamx494/pyscf/pyscf/scf/hf.py", line 277, in energy_tot
    e_tot = mf.energy_elec(dm, h1e, vhf)[0] + mf.energy_nuc()
  File "/home/gagliard/phamx494/pyscf/pyscf/pbc/scf/kuhf.py", line 152, in energy_elec
    e1 = 1./nkpts * np.einsum('kij,kji', dm_kpts[0], h1e_kpts)
  File "/home/gagliard/phamx494/anaconda/lib/python3.6/site-packages/numpy/core/einsumfunc.py", line 1228, in einsum
    return c_einsum(*operands, **kwargs)
ValueError: operands could not be broadcast together with remapped shapes [original->remapped]: (2,4,4)->(4,4,2) (3,4,4)->(4,4,3) 
```
The KRHF works just fine with the same GDF.
In fact, when I tried to put dms instead of dm_kpts using FFTDF:
`make_rho, nset, nao = ni._gen_rho_evaluator(cell, dms, hermi)`

I also got that error.

**Comment by gkc1000** [comment]
Ok, flagging as bug, will need someone to actually look into.

**Comment by sunqm** [comment]
Fixed in commit b494e1088d5c01c59a23b94aae51116154d5c69e
